### ISSUE TITLE
a11y in the Tab component 

### ISSUE THREAD

**Issue by majiddadashi** [salesforce/design-system-react#2780]
## Violating an a11y rule

The Tab component is violating an a11y rule ([this one](https://dequeuniversity.com/rules/axe/4.1/presentation-role-conflict?application=rocketvalidator)). This is also caught and indicated in the accessibility section of the Tabs storybook. 

### Functionality Issues:

- Keyboard navigation does not highlight the selected tab. 
- A screen-reader (I'm testing with Mac Voice Over) can not identify the number of Tabs and shows arbitrary behaviour.

### Fix

The part of code that needs fixing is [here](https://github.com/salesforce/design-system-react/blob/master/components/tabs/private/tab.jsx#L159-L177).

The tabindex should be moved to the child `<a>` tag as opposed to the `<li>` tag which has `role='presentation'`. Also, for arrow key navigation (to move from one Tab to another), we need to change the focus to the `<a>` tags as opposed to the `<li>` tag. Remember that role for the `<li>` tag is set to `presentation` so it's not in the a11y tree.

I'm sending a PR for this soon. Please let me know if I'm missing something or not on the right track.

**Comment by majiddadashi** [comment]
Related PR: #2781

**Comment by interactivellama** [comment]
Closing based on https://github.com/salesforce/design-system-react/issues/2780 Please re-open if not fixed.
